File: exercises/tests/balanced_test.rs
#[test]
fn balanced() {
    assert_eq!(skills::balanced::is_balanced("xxxyyy"), true);
}

#[test]
fn imbalanced() {
    assert_eq!(skills::balanced::is_balanced("xxxyyyy"), false);
}

#[test]
fn balanced_long() {
    assert_eq!(skills::balanced::is_balanced("yyxyxxyxxyyyyxxxyxyx"), true);
}

#[test]
fn imbalanced_long() {
    assert_eq!(skills::balanced::is_balanced("xyxxxxyyyxyxxyxxyy"), false);
}

#[test]
fn empty() {
    assert_eq!(skills::balanced::is_balanced(""), true);
}

#[test]
fn single() {
    assert_eq!(skills::balanced::is_balanced("x"), false);
}

File: exercises/tests/balanced.test.js
import test from "./test.js";
import balanced from "../src/balanced.js";

test({
  description: "balanced baseline",
  actual: balanced("xxxyyy"),
  expected: true,
});

test({
  description: "balanced basic imbalanced",
  actual: balanced("xxxyyyy"),
  expected: false,
});

test({
  description: "balanced long",
  actual: balanced("yyxyxxyxxyyyyxxxyxyx"),
  expected: true,
});

test({
  description: "balanced long imbalanced",
  actual: balanced("xyxxxxyyyxyxxyxxyy"),
  expected: false,
});

test({
  description: "balanced empty",
  actual: balanced(""),
  expected: true,
});

test({
  description: "balanced single character",
  actual: balanced("x"),
  expected: false,
});
